server/credentials.go

```diff
@@ -1,14 +1,95 @@
 package main
 
 import (
+	"fmt"
 	"runtime/debug"
 	"sort"
 	"strings"
 	"time"
 
 	"github.com/mattermost/mattermost-plugin-msteams/server/metrics"
+	"github.com/mattermost/mattermost-plugin-msteams/server/msteams/clientmodels"
 )
 
+const (
+	ResourceAccessTypeScope = "Scope"
+	ResourceAccessTypeRole  = "Role"
+)
+
+type expectedPermission struct {
+	Name           string
+	ResourceAccess clientmodels.ResourceAccess
+}
+
+// getResourceAccessKey makes a map key for the resource access that simplifies checking
+// for the resource access in question. (Technically, we could use the struct itself, but
+// this insulates us from unexpected upstream changes.)
+func getResourceAccessKey(resourceAccess clientmodels.ResourceAccess) string {
+	return fmt.Sprintf("%s+%s", resourceAccess.ID, resourceAccess.Type)
+}
+
+// describeResourceAccessType annotates the resource access type with the user facing term
+// shown in the Azure Tenant UI (Application vs. Delegated).
+func describeResourceAccessType(resourceAccess clientmodels.ResourceAccess) string {
+	switch resourceAccess.Type {
+	case ResourceAccessTypeRole:
+		return "Role (Application)"
+	case ResourceAccessTypeScope:
+		return "Scope (Delegated)"
+	default:
+		return resourceAccess.Type
+	}
+}
+
+// getExpectedPermissions returns the set of expected permissions, keyed by the
+// name the enduser would expect to see in the Azure tenant.
+func getExpectedPermissions() []expectedPermission {
```

Comment:
This is hard-coded for now, but I anticipate the permission set changing as we add functionality, some of which may be optional.